### DISCUSSION TITLE
Is there a "no wrap" option in `AnsiConsole`?

### DISCUSSION THREAD

**Discussion by nils-a** [spectreconsole/spectre.console#471] [Q&A]
While trying to fix [spectre1000](https://spectreconsole.net/analyzer/rules/spectre1000), I came across one small "problem":

I want to write some text to the console that can be easily copied to the clipboard. Hence, I don't really care for "readability" but more for "accuracy".

So, given the following code:
```cs
static void Main(string[] args)
        {
            var x = "some short words and then aReallyLongTextThatWillBeMoreThanTheLineLengthAndMightBeWrappedInSpectreConsoleWhichIsNotReallyWhatIWant";
            AnsiConsole.WriteLine(" -- ");
            AnsiConsole.WriteLine(x);
            AnsiConsole.WriteLine(" -- ");
            Console.WriteLine(x);
            AnsiConsole.WriteLine(" -- ");
        }
```

the output is

```text
 --
some short words and then                                                                                              
aReallyLongTextThatWillBeMoreThanTheLineLengthAndMightBeWrappedInSpectreConsoleWhichIsNotReallyWhatIWant
 --
some short words and then aReallyLongTextThatWillBeMoreThanTheLineLengthAndMightBeWrappedInSpectreConsoleWhichIsNotReall
yWhatIWant
 --
```

So, my question is: Is there a way to disable the word-breaking in `AnsiConsole`?

**Comment by patriksvensson** [comment]
Alas, there is no way of disbling this currently since it would break rendering. Let me think about it for a little while and I'll come back to you.

**Reply by vermiculus** [reply]
Thanks for your work on this project; it's incredibly valuable.

Our use case is echoing commands that we're running on behalf of the user, so line breaks are 'not ideal' – is this something that has been getting some noodle-time on how it might be addressed or worked-around?

Even if there was some sort of AnsiConsole primitive we could use to know if the terminal supports colors, we could branch off that and use `System.Color` for our purposes.

**Comment by scottbilas** [comment]
Here's a workaround I am using:

```
readonly struct SkipSpectreWordWrap : IDisposable
{
    readonly int _oldWidth = AnsiConsole.Profile.Width;

    public SkipSpectreWordWrap() => AnsiConsole.Profile.Width = 100000;
    public void Dispose() => AnsiConsole.Profile.Width = _oldWidth;
}

using var _ = new SkipSpectreWordWrap();
AnsiConsole.Markup(...);
```

Looking through the code, it seems that what we'd need is to set `RenderOptions.SingleLine=true`, but I don't see a way to do that, or an easy way to plumb it through. Perhaps it would be best done with a new markup token like `[option:+singleline]stuffthatwillnotbewrapped[/]`.

**Reply by patriksvensson** [reply]
@scottbilas The markup suggestion is an excellent idea!
